apps/src/lib/node/ledger/events/log/dumb_queries.rs

```diff
@@ -6,46 +6,59 @@
 //! tm.event='NewBlock' AND <accepted|applied>.<$attr>='<$value>'
 //! ```
 
-use namada::types::hash::HexEncodedHash;
+use std::marker::PhantomData;
+use namada::types::hash::Hash;
 
 use crate::node::ledger::events::{Event, EventType};
 
 /// A [`QueryMatcher`] verifies if a Namada event matches a
 /// given Tendermint query.
 #[derive(Debug, Clone)]
-pub struct QueryMatcher<'q> {
+pub struct QueryMatcher<'a, T>
+where
+    T: 'a +  PartialEq + TryFrom<&'a str>
+{
     event_type: EventType,
     attr: String,
-    value: &'q str,
+    value: T,
+    phantom: PhantomData<&'a T>,
 }
 
-impl<'q> QueryMatcher<'q> {
+impl<'a, T> QueryMatcher<'a, T>
+where
+    T: 'a + PartialEq + TryFrom<&'a str>
+{
     /// Checks if this [`QueryMatcher`] validates the
     /// given [`Event`].
-    pub fn matches(&self, event: &Event) -> bool {
+    pub fn matches(&self, event: &'a Event) -> bool {
         event.event_type == self.event_type
             && event
-                .attributes
-                .get(&self.attr)
-                .map(|value| value == self.value)
-                .unwrap_or_default()
+            .attributes
+            .get(&self.attr)
+            .map(|value| T::try_from(value.as_ref()).map(|v| v == self.value).unwrap_or(false))
```

Comment:
see this is the part that bothers me: to do these kind of checks, you need to decode the hex encoded string into a `Hash`, for every iteration over the event log

Reply:
Yeah, but had to do a utf8 conversion before instead.

Reply:
https://doc.rust-lang.org/stable/src/core/str/converts.rs.html#170-174
it just does a pointer cast, basically

Reply:
that's true because you were using the unchecked version.